src/features/device-state/utils.ts

```diff
@@ -33,10 +33,14 @@ export const serviceInstallFromImage = (
 	}
 
 	let id: number;
-	if (typeof image.is_a_build_of__service === 'object') {
-		id = image.is_a_build_of__service.__id;
+	if (Array.isArray(image.is_a_build_of__service)) {
+		id = image.is_a_build_of__service[0].id;
 	} else {
-		id = image.is_a_build_of__service;
+		id = image.is_a_build_of__service.__id;
+	}
+
+	if (isNaN(id)) {
```

Comment:
Do you want `isNaN` or `Number.isNaN` here?

Reply:
So, given that `isNaN()` will trap a wider range of conditions than `Number.isNaN()` it seems that we want `isNaN()`.  I'll give it a day or two and then mark this one as resolved unless I hear otherwise.